e2e/test/Helpers/TestDevice.cs

```diff
@@ -38,16 +39,17 @@ private TestDevice(Device device, Client.IAuthenticationMethod authenticationMet
         /// </summary>
         /// <param name="namePrefix">The prefix to apply to your device name</param>
         /// <param name="type">The way the device will authenticate</param>
-        public static async Task<TestDevice> GetTestDeviceAsync(string namePrefix, TestDeviceType type = TestDeviceType.Sasl)
+        public static async Task<TestDevice> GetTestDeviceAsync(TestLogger logger, string namePrefix, TestDeviceType type = TestDeviceType.Sasl)
```

Comment:
The calling test method will pass in it's `Logger` instance to all helper methods.